src/dbnode/storage/shard.go

```diff
@@ -774,6 +794,17 @@ func (s *dbShard) tickAndExpire(
 				if err != nil {
 					r.errors++
 				}
+				if opts.exportTopMetrics {
+					// TODO: find a cheaper way to get the metric name. 'Get()' iterates on all labels.
+					if metricNameBytes, ok := entry.Series.Metadata().Get([]byte(metricLabelName)); ok {
```

Comment:
I may little bit concern on iterating each metric series on each shard in this method. Maybe we can simply to do a profiling on this tick method to verify.  After each tick, I assume this will go through all shards in this method right?